.circleci/config.yml

```diff
@@ -57,20 +57,12 @@ jobs:
       - checkout
       - setup_remote_docker
 
-      - run:
-          name: Create a version.json
-          command: |
-            # create a version.json per https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
-            printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
-            "$CIRCLE_SHA1" \
-            "$CIRCLE_TAG" \
-            "$CIRCLE_PROJECT_USERNAME" \
-            "$CIRCLE_PROJECT_REPONAME" \
-            "$CIRCLE_BUILD_URL" > version.json
-
       - run:
           name: Build Docker image
-          command: docker build -t app:build .
+          command: | 
+            docker build -t app:build . \
+              --build-arg VERSION_BUILD_URL="$CIRCLE_BUILD_URL" \
+              --build-arg VERSION_TAG_NAME="$CIRCLE_TAG"
```

Comment:
I see in your summary this was done to make the version.json more faithful, but I don't understand why it was required in detail. What makes it more faithful rather running `go generate` in the ci job or leaving it as is?

Reply:
I should add: I'm extremely skeptical about including our entire git history into the dockerfile. The risks and resource use from that doesn't seem worth the rest of this.

Reply:
Regarding the resource usage, this shouldn't be a problem for `autograph-edge` because the Dockerfile doesn't actually retain the sources beyond the build stage (it just copies the binary into the last stage) so it doesn't matter all that much if the git directory hangs around during the build stage. This is different from the `autograph` container where we need the sources for all of the integration tests and IMO it tends to lead to bit of image bloat. But the edge container is extremely lean.

I suppose it could still be considered a risk or an inefficiency to leave the git history exposed during the build stage.

Reply:
I initially opted to go for this approach of running `go generate` inside the container because it means that we can eliminate all preprocessing steps prior to the build, in which case Docker's buildx can use a [git context](https://docs.docker.com/build/concepts/context/#git-repositories) directly for building. This ought to allow reproducible builds because the source can be tracked down to the git commit hash and all of the build state is captured in the build arguments.

The downside is that the container then needs access to the necessary bits from the git repository to generate a version.json on its own. This includes:
 - The commit hash (which we can get from `git rev-parse HEAD`)
 - The source URL (which we can get from `git remote get-url origin`)
 - The tag name being built (which generally has to be provided as a build-arg, because CI environments just provide the detached HEAD).
 - The build URL cannot be known from inside the container, so it has to be provided as an argument.

Reply:
However, this change isn't stricly necessary and we can reproduce the CircleCI approach by doing basically the same thing in Github actions.

I will revert this work and go that route instead.